# Key Performance Indicators (KPI) Analysis

## Objective

Calculate the key business metrics of the France Top 50 Playlist dataset. These KPIs summarize the dataset and provide the foundation for the Power BI dashboard.

In [2]:
import pandas as pd
import numpy as np

df = pd.read_csv("../data/france_top50_cleaned.csv")
print("Dataset shape: ",df.shape)
df.head()

Dataset shape:  (27780, 11)


,date,position,song,artist,popularity,duration_ms,album_type,total_tracks,is_explicit,album_cover_url,duration_minutes
0,2024-05-18,1,LA RUE,No Limit & Gazo & Damso,72,225137,single,1,True,https://i.scdn.co/image/ab67616d0000b27384f245...,3.75
1,2024-05-18,2,C'est carré le S,Naps,74,201533,album,22,False,https://i.scdn.co/image/ab67616d0000b27348f7c1...,3.36
2,2024-05-18,3,Freestyle LVL UP 1,Ninho,76,114182,single,1,True,https://i.scdn.co/image/ab67616d0000b273fbe613...,1.90
3,2024-05-18,4,Bolide allemand,SDM,79,176746,album,15,True,https://i.scdn.co/image/ab67616d0000b273455e89...,2.95
4,2024-05-18,5,Freestyle LVL UP 2,Ninho,71,149064,single,1,True,https://i.scdn.co/image/ab67616d0000b27351f1cb...,2.48


## KPI 1: Total Songs

### Objective

Calculate the total number of playlist entries available in the dataset.

In [3]:
total_songs = len(df)
print("Total Songs: ",total_songs)

Total Songs:  27780


## KPI 2: Total Unique Artists

### Objective

Calculate the total number of unique artists represented in the France Top 50 Playlist dataset.

In [4]:
total_artists = df["artist"].nunique()
print("Total Unique artists :",total_artists)

Total Unique artists : 324


## KPI 3: Average Popularity

### Objective

Calculate the average popularity score of songs in the France Top 50 Playlist dataset.

In [5]:
average_popularity = round(df["popularity"].mean(),2)
print("Average Popularity :",average_popularity)

Average Popularity : 76.65


## KPI 4: Explicit Song Percentage

### Objective

Calculate the percentage of explicit songs in the France Top 50 Playlist dataset.

In [6]:
explicit_percentage = round(df["is_explicit"].sum()/len(df)*100,2)
print("Explicit Song Percentage:", explicit_percentage, "%")

Explicit Song Percentage: 56.26 %


## KPI 5: Average Song Duration

### Objective

Calculate the average duration of songs in the France Top 50 Playlist dataset.

In [7]:
average_duration = round(df["duration_minutes"].mean(), 2)
print("Average Song Duration:", average_duration, "minutes")

Average Song Duration: 3.09 minutes


## KPI 6: Album vs Single Distribution

### Objective

Calculate the percentage distribution of album tracks and single releases in the France Top 50 Playlist dataset.

In [8]:
album_single_percentage = (
    df[df["album_type"].isin(["album", "single"])]["album_type"]
    .value_counts(normalize=True)
    .mul(100)
    .round(2)
)

print(album_single_percentage)

album_type
album     52.88
single    47.12
Name: proportion, dtype: float64


In [9]:
album_percentage = album_single_percentage["album"]
single_percentage = album_single_percentage["single"]

print("Album Percentage:", album_percentage, "%")
print("Single Percentage:", single_percentage, "%")

Album Percentage: 52.88 %
Single Percentage: 47.12 %


## KPI 7: Average Album Size

### Objective

Calculate the average number of tracks available in albums represented in the France Top 50 Playlist dataset.

In [10]:
average_album_size = round(df["total_tracks"].mean(), 2)
print("Average Album Size:", average_album_size)

Average Album Size: 9.0


## KPI 8: Most Frequent Artist

### Objective

Identify the artist with the highest number of appearances in the France Top 50 Playlist dataset.

In [11]:
artist_counts = df["artist"].value_counts()

most_frequent_artist = artist_counts.idxmax()
artist_appearances = artist_counts.max()

print("Most Frequent Artist:", most_frequent_artist)
print("Number of Appearances:", artist_appearances)

Most Frequent Artist: Werenoi
Number of Appearances: 2201


## KPI 9: Most Popular Song

### Objective

Identify the song with the highest popularity score in the France Top 50 Playlist dataset.

In [12]:
top_song = df.loc[df["popularity"].idxmax()]

print("Song:", top_song["song"])
print("Artist:", top_song["artist"])
print("Popularity:", top_song["popularity"])

Song: Paint The Town Red
Artist: Doja Cat
Popularity: 100


## KPI 10: Most Popular Artist

### Objective

Identify the artist with the highest average popularity score in the France Top 50 Playlist dataset.

In [13]:
artist_popularity = (
    df.groupby("artist")["popularity"]
      .mean()
      .round(2)
)

most_popular_artist = artist_popularity.idxmax()
highest_avg_popularity = artist_popularity.max()

print("Most Popular Artist:", most_popular_artist)
print("Average Popularity:", highest_avg_popularity)

Most Popular Artist: Jack Harlow
Average Popularity: 98.8


## KPI 11: Clean Content Dominance Ratio

### Objective

Calculate the percentage of clean (non-explicit) songs in the France Top 50 Playlist dataset.

In [14]:
clean_content_ratio = round(
    (df["is_explicit"].eq(False).sum() / len(df)) * 100, 2)
print("Clean Content Dominance Ratio:", clean_content_ratio, "%")

Clean Content Dominance Ratio: 43.74 %


## KPI 12: Content Acceptance Score

### Objective

Measure the overall audience acceptance of playlist content using the average popularity score.

In [15]:
content_acceptance_score = round(df["popularity"].mean(), 2)

print("Content Acceptance Score:", content_acceptance_score)

Content Acceptance Score: 76.65


# KPI Summary

## Key Findings

- The dataset contains **27,780** playlist entries from **324 unique artists**.
- The average song popularity is **76.65**, indicating that most tracks in the France Top 50 Playlist are highly popular.
- **56.26%** of the playlist consists of explicit songs, showing a slight dominance of explicit content.
- Songs have an average duration of **3.09 minutes**, reflecting current mainstream music trends.
- Album tracks (**52.88%**) slightly outnumber single releases (**47.12%**).
- The average album contains **9 tracks**.
- **Werenoi** is the most frequently featured artist with **2,201 appearances**.
- **Paint The Town Red** by **Doja Cat** achieved the highest popularity score (**100**).
- **Jack Harlow** has the highest average artist popularity score (**98.80**).